src/pal/src/map/virtual.cpp

```diff
@@ -138,7 +138,9 @@ namespace VirtualMemoryLogging
         IN BOOL result)
     {
         LONG i = InterlockedIncrement(&recordNumber) - 1;
-        LogRecord* curRec = (LogRecord*)&logRecords[i % MaxRecords];
+
+        // Converting to ULONG before modulo to ensure result is positive
+        LogRecord* curRec = (LogRecord*)&logRecords[(ULONG)i % MaxRecords];
```

Comment:
A nit - could you please change the `recordNumber`, `i` and `LogRecord::RecordId` to `ULONG` instead and cast the address passed to `InterlockedIncrement` and its result to `LONG`?  It would make inspecting the log a bit nicer.

Reply:
will do, and that makes sense!